operations/compare-helm-with-jsonnet/plugins/resolve-config/main_test.go

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func TestResolveConfigs(t *testing.T) {
+
+	t.Run("ResolveConfigs annotates values that match defaults", func(t *testing.T) {
+		querierDep, err := yaml.Parse(`
+apiVersion: v1
+kind: Deployment
+metadata:
+  name: querier
+  namespace: default
+spec:
+  template:
+    spec:
+      containers:
+        - name: mimir
+          image: grafana/mimir:latest
+          args:
+          - -target=querier
+`)
+		require.NoError(t, err)
+
+		mapper := NewConfigExtractor([]*yaml.RNode{querierDep})
+		configs, err := mapper.ResolveConfigs()
+		require.Len(t, configs, 1)
+		require.NoError(t, err)
+
+		require.Equal(t, "querier", configs[0].GetName(), "The MimirConfig metadata.name is set to the configured target")
+
+		target, err := configs[0].GetFieldValue("config.target")
+		require.NoError(t, err)
+		require.Equal(t, "querier", target, "The MimirConfig should have its target set to the configured target")
+
+		bindPort, err := configs[0].GetFieldValue("config.memberlist.bind_port")
+		require.NoError(t, err)
+		require.Equal(t, "7946 (default)", bindPort, "The MimirConfig should have its memberlist bind port set to the default value, with the text 'default'")
+	})
+
+	t.Run("ResolveConfigs sets configuration parameters from cli arguments", func(t *testing.T) {
```

Comment:
nit: this + the following 2 tests can be converted into a table test, they have the same assertions.